app/services/spine_service/reports/clinic/diagnosis_report.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module SpineService
+  module Reports
+    module Clinic
+      class DiagnosisReport
+        include ModelUtils
+        attr_reader :start_date, :end_date
+
+        def initialize(start_date:, end_date:, **kwargs)
+          @start_date = start_date.to_date.beginning_of_day.strftime('%Y-%m-%d %H:%M:%S')
+          @end_date = end_date.to_date.end_of_day.strftime('%Y-%m-%d %H:%M:%S')
+        end
+
+        def fetch_report
+          type = EncounterType.find_by_name 'Outpatient diagnosis'
+          data = Encounter.where('encounter_datetime BETWEEN ? AND ?
+            AND encounter_type = ?
+            AND obs.concept_id IN(6543, 6542)', @start_date, @end_date, type.id).\
+            joins('INNER JOIN obs ON obs.encounter_id = encounter.encounter_id
```

Comment:
It is best practice to do filters right on the joins. This will narrow done the search rather than returning the whole table

the obs concept that were being filtered above can be use here

```suggestion
INNER JOIN obs ON obs.encounter_id = encounter.encounter_id AND obs.concept_id IN (** your query that gets the concept ids based on the concept name**) AND obs.voided = 0
````

This goes for all the joins since these joins are not used ORM it best to add the scope manually. Thus table_name.voided/retired = 0